x-pack/test/functional/services/transform/transform_table.ts

```diff
@@ -190,25 +195,45 @@ export function TransformTableProvider({ getService }: FtrProviderContext) {
       });
     }
 
+    public async ensureDetailsOpen() {
+      await retry.tryForTime(30 * 1000, async () => {
+        if (!(await testSubjects.exists('transformExpandedRowTabbedContent'))) {
+          await testSubjects.click('transformListRowDetailsToggle');
+          await testSubjects.existOrFail('transformExpandedRowTabbedContent', { timeout: 1000 });
+        }
+      });
+    }
+
+    public async ensureDetailsClosed() {
+      await retry.tryForTime(30 * 1000, async () => {
+        if (await testSubjects.exists('transformExpandedRowTabbedContent')) {
+          await testSubjects.click('transformListRowDetailsToggle');
+          await testSubjects.missingOrFail('transformExpandedRowTabbedContent', { timeout: 1000 });
+        }
+      });
+    }
+
     public async assertTransformExpandedRow() {
-      await testSubjects.click('transformListRowDetailsToggle');
+      await retry.tryForTime(30 * 1000, async () => {
+        await this.ensureDetailsOpen();
```

Comment:
I'd suggest to move `ensureDetailsOpen` before the retry as it has its own 30 seconds retry.

Reply:
Done in https://github.com/elastic/kibana/pull/106990/commits/6cc323e18144841ca073d28195d1443d3c3b0fa3